Tasks/PackageDownloader/download.ts

```diff
@@ -0,0 +1,147 @@
+var fs = require('fs');
+var http = require('http');
+var DecompressZip = require('decompress-zip');
+var path = require('path')
+
+import * as restm from 'vso-node-api/RestClient';
+import * as vsom from 'vso-node-api/VsoClient';
+import * as tl from 'vsts-task-lib/task';
+import * as vsts from "vso-node-api/WebApi"
+
+const ApiVersion = "3.0-preview.1";
+
+async function main(): Promise<void> {
+	let feedId = tl.getInput("feed");
+	let packageId = tl.getInput("definition");
+	let version = tl.getInput("version");
+	let downloadPath = tl.getInput("downloadPath");
+	let collectionUrl = tl.getVariable("System.TeamFoundationCollectionUri");
+
+	var accessToken = getAuthToken();
+	var credentialHandler = vsts.getBearerHandler(accessToken);
+	var vssConnection = new vsts.WebApi(collectionUrl, credentialHandler);
+	var coreApi = vssConnection.getCoreApi();
+
+	await downloadPackage(coreApi, feedId, packageId, downloadPath, version);
+}
+
+function getAuthToken() {
+	var auth = tl.getEndpointAuthorization('SYSTEMVSSCONNECTION', false);
+	if (auth.scheme.toLowerCase() === 'oauth') {
+		return auth.parameters['AccessToken'];
+	}
+	else {
+		throw new Error(tl.loc("CredentialsNotFound"))
+	}
+}
+
+export async function downloadPackage(coreApi, feedId: string, packageId: string, downloadPath: string, version) {
+	var packageUrl = await getPackageUrl(coreApi.vsoClient, feedId, packageId);
+
+	await new Promise((resolve, reject) => {
+		coreApi.restClient.get(packageUrl, ApiVersion, null, { responseIsCollection: false }, async function (error, status, result) {
+			if (!!error && status != 200) {
+				reject(tl.loc("FailedToGetPackageUrl", error));
+			}
+
+			var packageType = result.protocolType.toLowerCase();
+			var packageName = result.name;
+
+			if (packageType == "nuget") {
+				var downloadUrl = await getDownloadUrl(coreApi.vsoClient, feedId, packageName, version)
+				
+				if (!tl.exist(downloadPath)) {
+					tl.mkdirP(downloadPath);
+				}
+
+				var zipLocation = path.join(downloadPath, packageName) + ".zip"
```

Comment:
Yes, one can download more than one package in a single release, in that case input downloadPath will be different for both packages, even if they are different versions of same package.